### PR TITLE
#121 Made Version 1 of Fuzzy Search

### PR DESCRIPTION
**Pull request by Siri2K**
- Should Perform searching on the product page

- Fuzzy search checks written string (does not have to be full text) and displays all matching products

### CODE REVIEW

File: amaznot/routes/web.php [hunk 1]
```diff
@@ -60,4 +61,7 @@
 Route::get('/orders/{id}', [OrderController::class, 'orderDetails'])->name('ordersitems');
 Route::get('/orders/{order_id}/{product_id}', [OrderController::class, 'productDetails'])->name('product');
 Route::post('/orders/deleteOrder', [OrderController::class, 'deleteOrder'])->name('deleteOrder');
-Route::post('/orders', [OrderController::class, 'store']);
\ No newline at end of file
+Route::post('/orders', [OrderController::class, 'store']);
+
+// Search Functions
+Route::get('/search/{search?}', [SearchController::class, 'search'])->name('search');
```
Comment (JamesPartsafas) [inline amaznot/routes/web.php:67]: Is the ? after the word search supposed to be there?

File: amaznot/app/Http/Controllers/SearchController.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use App\Models\Search;
+use Illuminate\Database\QueryException;
+
+class SearchController extends Controller
+{
+    public function search(Request $request, $search = "")
+    {
+        try
+        {
+            if($request->wantsJson())
```
Comment (JamesPartsafas) [inline amaznot/app/Http/Controllers/SearchController.php:15]: When is this if condition true? Also, if the condition is removed, the json you return throws an exception.

File: amaznot/app/Http/Controllers/SearchController.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Search;
+use Illuminate\Database\QueryException;
+
+class SearchController extends Controller
+{
+    public function search(Request $request, $search = "")
```
Comment (JamesPartsafas) [inline amaznot/app/Http/Controllers/SearchController.php:11]: If you're if condition returns false and there is no exception thrown, this controller method returns nothing at all. In fact, this seems to be the default behaviour, as going to /search/anything simply returns a blank page.

File: amaznot/routes/web.php (hunk 1 above)
Comment (Siri2K) [reply amaznot/routes/web.php:67]: Yeah, I'll send you the link, so u can fully confirm

File: amaznot/app/Http/Controllers/SearchController.php (hunk 3 above)
Comment (Siri2K) [reply amaznot/app/Http/Controllers/SearchController.php:11]: Oh damn, I'll fix it as soon as I can and send a new one

**Comment by Siri2K** [comment]
Looks good, but it fails the CI. I'll make some alteration so it passes the CI tests on Github

**Comment by Siri2K** [comment]
nvm, it still fails the CI test

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**